core/src/main/scala/tofu/Context.scala

```diff
@@ -20,48 +36,68 @@ trait Context[F[_]] {
   def extract[A](extr: Extract[Ctx, A]): WithContext[F, A] = new ContextExtractInstance[F, Ctx, A](this, extr)
 }
 
-object Context extends ContextInstances[HasContext] {
+object Context {
   def apply[F[_]](implicit ctx: Context[F]): HasContext[F, ctx.Ctx] = ctx
   type Aux[F[_], C] = HasContext[F, C]
 
-  def const[F[_]: Applicative, C](c: C): HasContext[F, C] = new Context[F] {
-    type Ctx = C
+  def const[F[_]: Applicative, C](c: C): WithContext[F, C] = new WithContext[F, C] {
     val functor: Functor[F] = Functor[F]
     val context: F[C]       = Applicative[F].pure(c)
   }
+
+  def make[F[_]: Functor, A](fa: F[A]): WithContext[F, A] = new WithContext[F, A] {
+    val functor: Functor[F] = implicitly
+    val context: F[A]       = fa
+  }
+
+  /** a mix-in for supplying environment data type companions with useful things */
+  trait Companion[C] {
+    type Has[F[_]] = WithContext[F, C]
+
+    implicit def tofuPromoteContextStructure[F[_], A](implicit
+        context: WithLocal[F, C],
+        field: C Contains A
+    ): WithLocal[F, A] =
+      context.subcontext(field)
+  }
 }
 
+/** lightweight version of ApplicativeAsk
+  *  Can be understood as functional replacement for global variable
+  * or implicit F[C]
+  */
 trait WithContext[F[_], C] extends Context[F] {
   override type Ctx = C
 }
 
-object WithContext extends ContextInstances[WithContext] {
+object WithContext {
   def apply[F[_], C](implicit ctx: WithContext[F, C]): WithContext[F, C] = ctx
-
 }
 
-trait ContextInstances[TC[f[_], r] >: WithContext[f, r]] extends LocalInstances[TC]
-
+/** lightweight version of ApplicativeLocal
+  * consider to use `WithLocal` or `In` for better type inference
```

Comment:
```suggestion
  * consider using `WithLocal` or `In` for better type inference
```
BTW, `type In[C, F[_]] = WithContext[F, C]` doesn't seem to be a proper alternative for `Local`, does it?

Reply:
Yes, an artifact of hasty copypasta